Problem with timestamps generated for CDC records in Scylla

Hello! I'm trying to do a Kafka to Scylladb sink by using Kafka Scylla Sink Connect but I got the following error:
`Caused by: com.datastax.driver.core.exceptions.InvalidQueryException: cdc: attempted to get a stream from an earlier generation than the currently used one. With CDC you cannot send writes with timestamps too far into the past, because that would break consistency properties (write timestamp: 1970/01/19 16:01:29, current generation started at: 2021/01/22 16:03:45)`

I have no idea from where is getting this timestamp with the date -1970/01/19.
The ScyllaDB cluster is on VM, the Kafka Broker is running on another VM and the Kafka Connect is running in a Kubernetes container. 

What I guess is causing the issue is the server time or at least the clock that Kafka Connect's Scylla client is using so I've checked the server times on the VM and in the Container and they seem to be okay (keep in mind it was needed some time between checking the server time on the different targets):
Dev Scylla VMs server time          ->  Wed 2021-02-10 14:17:19 UTC
Dev Kafka VMs server time          ->  Wed 2021-02-10 14:26:10 UTC
Dev K8s Kafka Pods  server time  ->  Wed 2021 Feb 10 14:20:09 UTC

Is there anything that can fix this? I didn't found any solution on the internet neither any Configuration Property which to do anything with that.

--- a/src/main/java/io/connect/scylladb/ScyllaDbSinkTaskHelper.java
+++ b/src/main/java/io/connect/scylladb/ScyllaDbSinkTaskHelper.java
@@ -95,7 +95,9 @@ public class ScyllaDbSinkTaskHelper {
       boundStatement.setDefaultTimestamp(topicConfigs.getTimeStamp());
     } else {
       boundStatement.setConsistencyLevel(this.scyllaDbSinkConnectorConfig.consistencyLevel);
-      boundStatement.setDefaultTimestamp(record.timestamp());
+      // Timestamps in Kafka (record.timestamp()) are in millisecond precision,
+      // while Scylla expects a microsecond precision: 1 ms = 1000 us.
+      boundStatement.setDefaultTimestamp(record.timestamp() * 1000);
     }
     return boundStatement;
   }

--- a/src/main/java/io/connect/scylladb/topictotable/TopicConfigs.java
+++ b/src/main/java/io/connect/scylladb/topictotable/TopicConfigs.java
@@ -140,7 +140,9 @@ public class TopicConfigs {
   }
 
   public void setTtlAndTimeStampIfAvailable(SinkRecord record) {
-    this.timeStamp = record.timestamp();
+    // Timestamps in Kafka (record.timestamp()) are in millisecond precision,
+    // while Scylla expects a microsecond precision: 1 ms = 1000 us.
+    this.timeStamp = record.timestamp() * 1000;
     if (timeStampMappedField != null) {
       Object timeStampValue = getValueOfField(record.value(), timeStampMappedField);
       if (timeStampValue instanceof Long) {

